eclass/tests/flag-o-matic.sh

```diff
@@ -146,4 +146,74 @@ out=$(CC=clang test-flags-CC -finline-limit=1200)
 ftend
 fi
 
+clear-env() {
+	unset $(all-flag-vars)
+}
+
+fom-tbegin() {
+	clear-env
```

Comment:
Couldn't you just `clear-env` once and limit the scope of `CFLAGS=` being set for calls?